### PR TITLE
TooltipComponent related mappings

### PR DESCRIPTION
**Pull request by emilyploszaj**

### CODE REVIEW

**Review by i509VCB — CHANGES_REQUESTED** [review]
Please add some javadoc, I had no damn clue what a TooltipComponent is from discussion on discord.

File: mappings/net/minecraft/client/gui/screen/TooltipComponent.mapping [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/class_5684 net/minecraft/client/gui/screen/TooltipComponent
+	METHOD method_32661 getWidth ()I
+	METHOD method_32662 createOrderedTextTooltipComponent (Lnet/minecraft/class_5481;)Lnet/minecraft/class_5684;
+	METHOD method_32663 createTooltipComponent (Lnet/minecraft/class_5632;)Lnet/minecraft/class_5684;
```
Comment (YanisBft) [inline mappings/net/minecraft/client/gui/screen/TooltipComponent.mapping]: This creates what you called a BundleTooltipComponent

**Comment by YanisBft** [comment]
Also the classes don't really fit in nm.client.gui.screen imo

File: mappings/net/minecraft/client/gui/screen/TooltipComponent.mapping (hunk 1 above)
Comment (emilyploszaj) [reply mappings/net/minecraft/client/gui/screen/TooltipComponent.mapping]: Yes but the structure of the method suggests it's for any generic `TooltipData` and would be used if mojang implemented any more.

**Review by jaskarth — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

File: mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_5683 net/minecraft/client/gui/screen/OrderedTextTooltip
```
Comment (Juuxel) [inline mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping]: Should this also have `Component` as a suffix like `BundleTooltipComponent`?

**Comment by haykam821** [comment]
If these are `TooltipComponent`s then should `Text` be a `TextComponent`?

**Comment by Juuxel** [comment]
No, as these are quite literally parts of a tooltip that are not tooltips on their own, unlike `Text`s which *are* texts even when not combined.

File: mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping (hunk 2 above)
Comment (emilyploszaj) [reply mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping]: Yep, that was a typo, it's been adjusted

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_5683 net/minecraft/client/gui/screen/OrderedTextTooltipComponent
```
Comment (YanisBft) [inline mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping]: You didn't rename the file

**Review by YanisBft — COMMENTED** [review]
About the packages, maybe the 3 tooltip component classes could go to a new `nm.client.gui.tooltip` package?

File: mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping (hunk 3 above)
Comment (emilyploszaj) [reply mappings/net/minecraft/client/gui/screen/OrderedTextTooltip.mapping]: heh, oops, fixed now

**Comment by emilyploszaj** [comment]
That sounds like a good package, I've moved the files in question

**Review by YanisBft — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

**Pull request merged**